sdk/trace/span.go

```diff
@@ -210,7 +210,7 @@ func (s *recordingSpan) SetStatus(code codes.Code, description string) {
 // attributes the span is configured to have, the last added attributes will
 // be dropped.
 func (s *recordingSpan) SetAttributes(attributes ...attribute.KeyValue) {
-	if !s.IsRecording() {
+	if !s.IsRecording() || len(attributes) == 0 {
```

Comment:
The fixes in this file appear unrelated to the once/oncefunc change that describe this PR.
Could you split them into another PR? (and test the issues they aim to fix?)

Reply:
Hey @dmathieu 
Thanks for reviewing! I removed the unrelated commits and moved them to https://github.com/open-telemetry/opentelemetry-go/pull/5864
Have a great day.

Reply:
Did you update the benchmarks as well so they only include the PR's changes?

Reply:
Yes I changed the PR description to just include https://github.com/open-telemetry/opentelemetry-go/pull/5858/commits/e3c5cd17d06d2c6726042483dcffc709f4502b91